pandas/io/parquet.py

```diff
@@ -48,6 +48,33 @@ def get_engine(engine: str) -> "BaseImpl":
     raise ValueError("engine must be one of 'pyarrow', 'fastparquet'")
 
 
+def _get_filepath(
+    path: FilePathOrBuffer,
+    fs: Any,
+    storage_options: StorageOptions = None,
+    mode: str = "rb",
+    is_dir: bool = False,
+) -> Tuple[IOHandles, Any]:
+    """File handling for PyArrow."""
+    path = stringify_path(path)
+    if is_fsspec_url(path) and fs is None:
+        fsspec = import_optional_dependency("fsspec")
+
+        fs, path = fsspec.core.url_to_fs(path, **(storage_options or {}))
+    elif storage_options:
+        raise ValueError("storage_options passed with buffer or non-fsspec filepath")
+
+    # (ab-)use IOHandles to use its context manager
+    handles = IOHandles(handle=cast(Buffer, path), compression={"method": None})
```

Comment:
`path` is not a Buffer here?

Reply:
yes, it isn't a buffer. I was trying to abuse IOHandles as it can be used as context manager. I can change it to return a tuple of path, fs, Optional[IOHandles] and then have a try-finally to ensure that opened file handles are still closed in case there are errors while reading/writing